pkg/jobs/registry.go

```diff
@@ -1341,10 +1341,13 @@ func (r *Registry) DeleteTerminalJobByID(ctx context.Context, id jobspb.JobID) e
 			if err != nil {
 				return err
 			}
-			_, err = txn.Exec(
-				ctx, "delete-job-info", txn.KV(), "DELETE FROM system.job_info WHERE job_id = $1", id,
-			)
-			return err
+			if r.settings.Version.IsActive(ctx, clusterversion.V23_1CreateSystemJobInfoTable) {
+				_, err := txn.Exec(
+					ctx, "delete-job-info", txn.KV(), "DELETE FROM system.job_info WHERE job_id = $1", id,
+				)
+				return err
```

Comment:
very minor style nit: I find early returns like this easy to mistake for the standard go boilerplate `return err` err handling when skimming. I might elect to invert and do `!IsActive{ return nil }` instead, or make this `if err != nil { return err}` and fall through to the `return nil` below when not-nil

Reply:
Nits always welcome. Went with the second suggestion.